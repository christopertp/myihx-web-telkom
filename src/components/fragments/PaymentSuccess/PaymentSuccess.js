import React, { useEffect } from 'react';
import { useLocation } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import Button from '../../elements/Button';
import Heading from '../../elements/Heading';
import IconButton from '../../elements/IconButton';
import { thousand } from '../../../utils/format';
import styles from './styles.scoped.css';

export default function PaymentSuccess() {
  const dispatch = useDispatch();
  const query = new URLSearchParams(useLocation().search);
  const invoice = `MYIRX-${query.get('invoice')}`;
  const { productInfo, customer,
    fetchPaymentStatus, totalDue } = useSelector(s => s.payment);

  useEffect(() => {
    invoice && dispatch(fetchPaymentStatus(invoice));
  }, [invoice]);

  return (
    <section className={styles.cardroot}>
      <IconButton className={styles.back} name="close_red" to="/" />
      <header className={styles.headerDetail}>
        <img alt="success" className={styles.img} src="/assets/grfx_success.svg"/>
        <h1>Payment Success</h1>
        <p>Friday, 5th June 2020</p>
        <p>Order ID {invoice}</p>
      </header>
      <article>
        <Heading>Package</Heading>
        <section className={styles.detailPackage}>
          <article>
            <span>
              <h4>{productInfo.value}</h4>
              <p>{productInfo.unit}</p>
            </span>
            <h4>{productInfo.name}</h4>
            <small>{productInfo.description}</small>
          </article>
          <article>
            <h4>Rp{thousand(totalDue)}<small>/month</small></h4>
            <small>Billing starts the month after installation</small>
          </article>
        </section>
        <Heading>Your Information</Heading>
        <section className={styles.information}>
          <article>
            <h4>Name</h4>
            <p>{customer.name}</p>
          </article>
          <article>
            <h4>KTP Number</h4>
            <p>{customer.identityNumber}</p>
          </article>
          <article>
            <h4>Installation Address</h4>
            <p>{customer.installationAddress}</p>
          </article>
        </section>
      </article>
      <article className={styles.nextSchedule}>
        <h4>Next, schedule your internet installation below</h4>
        <Button to="/installation">SCHEDULE INSTALLATION</Button>
      </article>
    </section>
  );
}
