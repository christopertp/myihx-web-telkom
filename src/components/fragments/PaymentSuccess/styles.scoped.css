.back {
  height: 3rem;
  padding: 0.5rem 0.5rem 0.5rem 0;
}

.cardroot{
  background-color: #fff;
  border-radius: 0.5rem;
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  color: #333435;
  margin: auto;
  max-width: 45rem;
  padding: 1.5rem;
}

.detailPackage {
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 1, 0.15);
  border-radius: 0.5rem;
  margin-bottom: 1.5rem;
  margin-top: 0.7rem;
  overflow: hidden;
}

.detailPackage > article:first-child {
  background: linear-gradient(0deg, #f3f3f3 0%, #fff 65%);
  display: grid;
  grid-template-columns: auto 1fr;
  padding: 1rem;
}

.detailPackage > article:first-child > span {
  align-items: center;
  background: linear-gradient(135deg, #6a120c, #b21f15, #ee4c24, #ff7e00);
  border-radius: 0.5rem;
  color: #ffffff;
  display: inline-flex;
  flex-direction: column;
  grid-row: 1/3;
  height: 4.375rem;
  justify-content: center;
  margin: auto 1rem auto 0;
  width: 4.375rem;
}

.detailPackage > article:first-child > span > h4 {
  font-size: 2rem;
  line-height: 1.2;
}

.detailPackage > article:first-child > span > p {
  color: rgba(255, 255, 255, 0.5);
  font-size: 0.8rem;
  text-transform: uppercase;
}

.detailPackage > article:first-child > h4 {
  font-size: 1.3rem;
  margin-top: auto;
}

.detailPackage > article:first-child > small {
  font-size: 1rem;
  margin-bottom: auto;
}

.detailPackage > article:last-child {
  padding: 1rem;
}

.detailPackage > article:last-child > h4 {
  font-size: 1.5rem;
}

.detailPackage > article:last-child > h4 > small {
  color: #b2b4b5;
  font-size: 1rem;
  font-weight: 400;
}

.detailPackage > article:last-child > small {
  color: #66686a;
  font-size: 1rem;
  font-weight: 400;
}

.headerDetail{
  text-align: center;
  font-size: 0.75rem;
}

.headerDetail > h1 {
  color: #333435;
  font-size: 1.25rem;
}

.img {
  width: 12.5rem;
}

.information{
  margin-top: 0.5rem;
}

.information > article {
  margin-bottom: 1.25rem;
}

.information > article > h4 {
  font-weight: 500;
  line-height: 1rem;
  margin-bottom: 0.5rem;
}

.information > article > h4,
.headerDetail {
  color: #66686a;
}

.load {
  align-items: center;
  display: flex;
  justify-content: center;
  margin: auto;
  width: 45rem;
}

.nextSchedule > h4 {
  color: #66686a;
  font-weight: 400;
  margin-bottom: 0.625rem;
  text-align: center;
}

@media only screen and (max-width: 48em) {
  .cardroot {
    background-color: transparent;
    box-shadow: none;
    min-width: 100%;
    padding: 0;
  }

  .headerDetail {
    margin-bottom: 0.5rem;
  }
  
  .information {
    flex-direction: column;
  }

  .information > article {
    margin: 0.6rem 0;
  }

  .information > article > h4,
  .headerDetail > p {
    font-size: 0.75rem;
  }
  
  .detailPackage > article:first-child > span {
    height: 3.75rem;
    margin-right: 0.5rem;
    width: 3.75rem;
  }

  .detailPackage > article:first-child > span > h4 {
    font-size: 1.5rem;
  }

  .detailPackage > article:first-child > span > p {
    font-size: 0.7rem;
  }
  
  .detailPackage > article:first-child > h4 {
    padding-top: 0.3rem;
  }

  .detailPackage > article:first-child > small {
    font-size: 0.75rem;
  }
  
  .detailPackage > article:last-child > h4 {
    font-size: 1.25rem;
  }
  
  .detailPackage > article:last-child > small {
    font-size: 0.75rem;
  }

  .information > article > p {
    font-size: 0.9375rem;
  }

  .headerDetail {
    text-align: left;
  }

  .headerDetail > h1 {
    font-size: 1.5rem;
  }

  .headerDetail > p {
    line-height: 1.125rem;
  }

  .img {
    margin-bottom: 1rem;
    width: 6.25rem;
  }

  .load {
    height: 100vh;
    margin-bottom: 6rem;
    width: auto;
  }
  
  .nextSchedule > h4 {
    font-size: 0.75rem;
  }
}
