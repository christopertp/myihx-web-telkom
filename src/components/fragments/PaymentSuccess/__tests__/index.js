import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';
import PaymentSuccess from '../PaymentSuccess';

useSelector.mockImplementation(fn => {
  fn({});
  return {
    fetchPaymentStatus: jest.fn(),
    customer: {
      name: 'squad4test2',
      identityType: 'ktp',
      identityNumber: '123123',
      installationAddress: 'Jl. Bintaro Puspita Raya Blok A No.9, RT.9/RW.2, Pesanggrahan, Kec. Pesanggrahan, Kota Jakarta Selatan, Daerah Khusus Ibukota Jakarta 12320, Indonesia',
      province: 'Bali',
      cityName: 'DENPASAR',
      postCode: '80222',
    },
    deadline: '2020-06-27T03:00:00.043Z',
    paymentStatus: 'SUCCESS',
    paymentType: 'DEPOSIT',
    productInfo: {
      value: 50,
      unit: 'Mbps',
      name: 'Indihome Fiber Triple Play Prestige',
      description: '50 Mbps, UseeTV Essential, Free Local SLJJ 1000 menit, wifi.id Seamless, Cloud Storage 8Gb, Movin Phone, HOOQ, Iflix',
      productDetails: [{
        title: 'First month deposit',
        amount: 825000,
      },
      {
        title: 'Tax 10%',
        amount: 82500,
      }],
    },
    totalDue: 907500,
  };
});

describe('src/components/fragments/PaymentSuccess', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<PaymentSuccess />);
    expect(tree).toMatchSnapshot();
  });

  test('call get payment status', () => {
    const dispatch = jest.fn();
    useDispatch.mockImplementationOnce(() => dispatch);

    PaymentSuccess();
    expect(dispatch).toHaveBeenCalled();
  });
});
