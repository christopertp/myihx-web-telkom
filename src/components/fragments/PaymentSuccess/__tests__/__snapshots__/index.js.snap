// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fragments/PaymentSuccess render 1`] = `
<section>
  <IconButton
    className=""
    name="close_red"
    onClick={[Function]}
    title=""
    to="/"
    type="button"
  />
  <header>
    <img
      alt="success"
      src="/assets/grfx_success.svg"
    />
    <h1>
      Payment Success
    </h1>
    <p>
      Friday, 5th June 2020
    </p>
    <p>
      Order ID 
      MYIRX-null
    </p>
  </header>
  <article>
    <Heading
      className=""
    >
      Package
    </Heading>
    <section>
      <article>
        <span>
          <h4>
            50
          </h4>
          <p>
            Mbps
          </p>
        </span>
        <h4>
          Indihome Fiber Triple Play Prestige
        </h4>
        <small>
          50 Mbps, UseeTV Essential, Free Local SLJJ 1000 menit, wifi.id Seamless, Cloud Storage 8Gb, Movin Phone, HOOQ, Iflix
        </small>
      </article>
      <article>
        <h4>
          Rp
          907.500
          <small>
            /month
          </small>
        </h4>
        <small>
          Billing starts the month after installation
        </small>
      </article>
    </section>
    <Heading
      className=""
    >
      Your Information
    </Heading>
    <section>
      <article>
        <h4>
          Name
        </h4>
        <p>
          squad4test2
        </p>
      </article>
      <article>
        <h4>
          KTP Number
        </h4>
        <p>
          123123
        </p>
      </article>
      <article>
        <h4>
          Installation Address
        </h4>
        <p>
          Jl. Bintaro Puspita Raya Blok A No.9, RT.9/RW.2, Pesanggrahan, Kec. Pesanggrahan, Kota Jakarta Selatan, Daerah Khusus Ibukota Jakarta 12320, Indonesia
        </p>
      </article>
    </section>
  </article>
  <article>
    <h4>
      Next, schedule your internet installation below
    </h4>
    <Button
      className=""
      disabled={false}
      onClick={null}
      to="/installation"
      type="button"
      variant="red"
    >
      SCHEDULE INSTALLATION
    </Button>
  </article>
</section>
`;
